### PR TITLE
open-cluster-management/backlog#7725-- new git connections

### PR DESCRIPTION
**Pull request by swopebe**
@rokej I think we have this pretty straight-forward by now, but when you are ready to review, you can comment or commit. Thanks!

### CODE REVIEW

**Review by rokej — COMMENTED** [review]
Can the content and sub-topic names be the same or similar to what I provided in https://github.com/open-cluster-management/backlog/issues/7725#issue-766827230 ?

The first introduction paragraph just needs to highlight that there are several different ways to configure connection to Git server. Then the sub-topics provides instructions and examples of different connection methods.

**Comment by swopebe** [comment]
@rokej I will work on them -- they all need to start with a verb and be parallel and not too long. I will make a few changes to satisfy that and your request.

**Comment by swopebe** [comment]
@rokej I made some updates, review when you have a moment? Thanks!

File: manage_applications/connect_git_repository.adoc [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+  namespace: channel-ns
+spec:
+    type: Git
+    pathname: <Git HTTPS URL>
+    secretRef:
+      name: my-git-secret
+----
+
+[#connecting-to-an-http-server]
+== Connecting to an insecure HTTPS connection
```
Comment (rokej) [inline manage_applications/connect_git_repository.adoc]: Making an insecure HTTPS connection to Git server

File: manage_applications/connect_git_repository.adoc [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+  namespace: channel-ns
+spec:
+  configMapRef:
+    name: git-ca
+  pathname: <Git HTTPS URL>
+  type: Git
+----
+
+[#git-SSH-connection]
+== Connecting to a Git SSH connection
```
Comment (rokej) [inline manage_applications/connect_git_repository.adoc]: Making an SSH connection to Git server

File: manage_applications/connect_git_repository.adoc [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+spec:
+  secretRef:
+    name: git-ssh-key
+  pathname: <Git SSH URL>
+  type: Git
+  insecureSkipVerify: true
+----
+
+[#update-channel-secret-config]
+== Updating channel secret and config map
```
Comment (rokej) [inline manage_applications/connect_git_repository.adoc]: Updating certificates and SSH keys

File: manage_applications/connect_git_repository.adoc [hunk 4]
```diff
@@ -0,0 +1,273 @@
+
+[#connecting-git-repository]
+= Connecting a Git repository using multiple methods
```
Comment (rokej) [inline manage_applications/connect_git_repository.adoc]: Configuring application channel and subscription for secure Git connection

File: manage_applications/connect_git_repository.adoc [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+
+Git channels and subscriptions connect to the specified Git repository HTTPS and expect the Git server SSL certificate to be issued by a known CA. If the certificate is issued by an unknown CA, such as a self-signed certificate, you want to add a custom CA for a Git channel that is secure.
```
Comment (rokej) [inline manage_applications/connect_git_repository.adoc]: Git channels and subscriptions connect to the specified Git repository through HTTPS or SSH. The following application channel configurations can be used for secure Git connections.

**Review by rokej — APPROVED** [review]

**Pull request merged**